### PR TITLE
GODRIVER-2331 Replace collStats with count

### PR DESCRIPTION
**Pull request by prestonvasquez**
GODRIVER-2331

This ticket replaces the `collStats` operation in the `Count` executor with `count`.

### CODE REVIEW

**Review by qingyang-hu — APPROVED** [review]
👍 LGTM

File: mongo/integration/unified/database_operation_execution.go [hunk 1]
```diff
@@ -12,12 +12,65 @@ import (
...
+
+		switch key {
+		case "collection":
+			collName = val.StringValue()
+		case "pipeline":
+			pipeline = testhelpers.RawToInterfaceSlice(val.Array())
+		case "viewOn":
+			viewOn = val.StringValue()
+		default:
+			return nil, fmt.Errorf("unrecognized createCollection option %q", key)
```
Comment (benjirewis) [inline mongo/integration/unified/database_operation_execution.go:49]:
```suggestion
			return nil, fmt.Errorf("unrecognized createView option %q", key)
```
[opt] I guess `executeCreateView` is run through the `createCollection` operation, so maybe the current message is more accurate?

File: mongo/integration/unified/database_operation_execution.go [hunk 2]
```diff
@@ -65,7 +118,7 @@ func executeCreateCollection(ctx context.Context, operation *operation) (*operat
 		}
 	}
 	if collName == "" {
-		return nil, newMissingArgumentError("collName")
+		return nil, newMissingArgumentError("collection")
```
Comment (benjirewis) [inline mongo/integration/unified/database_operation_execution.go:121]: Nice catch.

File: vendor/github.com/google/go-cmp/cmp/path.go [hunk 3]
```diff
@@ -315,7 +315,7 @@ func (tf Transform) Option() Option { return tf.trans }
 // pops the address from the stack. Thus, when traversing into a pointer from
 // reflect.Ptr, reflect.Slice element, or reflect.Map, we can detect cycles
 // by checking whether the pointer has already been visited. The cycle detection
-// uses a seperate stack for the x and y values.
+// uses a separate stack for the x and y values.
```
Comment (benjirewis) [inline vendor/github.com/google/go-cmp/cmp/path.go:318]: Hmm any reason to touch the `vendor` file?

Comment (prestonvasquez) [reply vendor/github.com/google/go-cmp/cmp/path.go:318]: Oh good catch, I didn't realize that was a vendor file.

**Review by benjirewis — APPROVED** [review]
Looks good! Great work.

**Pull request merged**